server/internal/usecase/interactor/asset.go

```diff
@@ -137,32 +143,41 @@ func (i *Asset) Create(ctx context.Context, inp interfaces.CreateAssetParam, op
 				return nil, err
 			}
 
-			taskPayload := task.DecompressAssetPayload{
-				AssetID: a.ID().String(),
-				Path:    a.RootPath(),
-			}
-			if err := i.gateways.TaskRunner.Run(ctx, taskPayload.Payload()); err != nil {
-				return nil, err
-			}
-
-			a.UpdateArchiveExtractionStatus(lo.ToPtr(asset.ArchiveExtractionStatusInProgress))
-			if err := i.repos.Asset.Save(ctx, a); err != nil {
-				return nil, err
-			}
-
-			if err := i.event(ctx, Event{
-				Workspace: prj.Workspace(),
-				Type:      event.AssetCreate,
-				Object:    a,
-				Operator:  op.Operator(),
-			}); err != nil {
-				return nil, err
+			if !inp.SkipDecompression {
+				if err := i.triggerDecompressEvent(ctx, a, prj, op); err != nil {
+					return a, err
+				}
 			}
 
 			return a, nil
 		})
 }
 
+func (i *Asset) triggerDecompressEvent(ctx context.Context, a *asset.Asset, prj *project.Project, op *usecase.Operator) error {
+	taskPayload := task.DecompressAssetPayload{
+		AssetID: a.ID().String(),
+		Path:    a.RootPath(),
+	}
+	if err := i.gateways.TaskRunner.Run(ctx, taskPayload.Payload()); err != nil {
+		return err
+	}
+
+	a.UpdateArchiveExtractionStatus(lo.ToPtr(asset.ArchiveExtractionStatusInProgress))
+	if err := i.repos.Asset.Save(ctx, a); err != nil {
+		return err
+	}
+
+	if err := i.event(ctx, Event{
```

Comment:
move to original usecase function because it should be called even if decompression is skipped